src/main/java/com/zufar/onlinestore/payment/endpoint/PaymentEndpoint.java

```diff
@@ -38,36 +36,63 @@ public class PaymentEndpoint {
     private final PaymentApi paymentApi;
 
     @PostMapping
-    public ResponseEntity<PaymentDetailsWithTokenDto> createPayment(@RequestBody @Valid final CreatePaymentDto createPaymentDto) throws PaymentIntentProcessingException {
-        PaymentDetailsWithTokenDto createdPayment = paymentApi.createPayment(createPaymentDto);
+    public ResponseEntity<ApiResponse<ProcessedPaymentWithClientSecretDto>> processPayment(@RequestParam @NotEmpty final String cardDetailsTokenId) {
+        ProcessedPaymentWithClientSecretDto processedPayment = paymentApi.processPayment(cardDetailsTokenId);
+
+        ApiResponse<ProcessedPaymentWithClientSecretDto> apiResponse = ApiResponse.<ProcessedPaymentWithClientSecretDto>builder()
+                .data(processedPayment)
+                .messages(List.of("Payment successfully processed"))
```

Comment:
I think we should use messages only for errors. Just leave messages empty as you have data field filled by `processedPayment`

Reply:
Done